ghcide/src/Development/IDE/Core/ProgressReporting.hs

```diff
@@ -33,119 +34,156 @@ import           Language.LSP.Server            (ProgressAmount (..),
                                                  withProgress)
 import qualified Language.LSP.Server            as LSP
 import qualified StmContainers.Map              as STM
-import           UnliftIO                       (Async, async, cancel)
+import           UnliftIO                       (Async, MonadUnliftIO, async,
+                                                 bracket, cancel)
 
 data ProgressEvent
-    = KickStarted
-    | KickCompleted
+  = ProgressStarted
+  | ProgressCompleted
```

Comment:
Can we just get rid of `progressStop`, given that it just does the same thing as `progressUpdate ProgressCompleted`?

Reply:
`Kick` progress tracking working differently than `indexHIeFile`.

HLS begins: Progress tracking started

loop:
  Kick start ProgressStarted
  Kick end ProgressCompleted

HLS begins: Progress tracking Stop

Reply:
Oh yes, I missed that the state update is different, `ProgressCompleted` goes back to `NotStarted`. It's not clear that that gets us much, though :thinking:

Reply:
`NotStarted` and `Stopped` are the same state, except you can't start again from `Stopped`. But why does that even matter? In both cases you stop the session, it's not like `Stopped` cleans up any more resources!

Reply:
So the way it is structured we get a restartable progress tracker, which is a reasonable thing to have when you keep a single reference to it alive for a long time.. But I think it could be simpler.

Reply:
> So the way it is structured we get a restartable progress tracker

Yes, I think it is intended for multiple `Kick`'s.

Reply:
or we need to initialize progress tracker at each kick.

Reply:
I guess, `Stopped ` state is used to forbit us to start the progress tracker again after the we shutdown hls

Reply:
That doesn't seem actually useful, though. I'd be inclined to remove it.